pandas/core/generic.py

```diff
@@ -2146,8 +2146,17 @@ def to_excel(self, excel_writer, sheet_name="Sheet1", na_rep="",
                  index_label=None, startrow=0, startcol=0, engine=None,
                  merge_cells=True, encoding=None, inf_rep="inf", verbose=True,
                  freeze_panes=None):
+
         df = self if isinstance(self, ABCDataFrame) else self.to_frame()
 
+        max_rows = 2**20
+        max_cols = 2**14
+        num_rows, num_cols = df.shape
+        if num_rows > max_rows or num_cols > max_cols:
+            raise ValueError(f"This sheet is too large! Your sheet size is: "
+                             f"{(num_rows, num_cols)}. "
+                             f"Max sheet size is: {(max_rows, max_cols)}.")
+
```

Comment:
We still support 3.5, so f-strings cannot be used unfortunately.

Reply:
make max_rows, max_cols class level variables and make them a more descriptive name

Reply:
@gfyoung will switch to .format
@jreback I chose not to make them class level variables, because the size constraints only apply when writing to excel and not for the object in general.

Reply:
not sure you saw my comment; these size checks need to be in the excel.py (all of them); then these are simply class variables of the ExcelFile